### PR TITLE
[Identity] Validate identity config for MICredential

### PR DESCRIPTION
**Pull request by pvaneck**
Closes: https://github.com/Azure/azure-sdk-for-python-pr/issues/920

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py [hunk 1]
```diff
@@ -17,6 +17,27 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def validate_identity_config(client_id: Optional[str], identity_config: Optional[Mapping[str, str]]) -> None:
+    if identity_config:
+        if client_id:
+            if any(key in identity_config for key in ("object_id", "resource_id")):
```
[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py:23]: nevermind

File: sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py [hunk 2]
```diff
@@ -17,6 +17,27 @@
...
+                raise ValueError(
+                    "identity_config must not contain 'object_id' or 'resource_id' when 'client_id' is present"
+                )
+            if "client_id" in identity_config and identity_config["client_id"] != client_id:
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py:27]: I don't think we need to validate if they are same or not. As long as client_id is specified in two places, we raise.

File: sdk/identity/azure-identity/azure/identity/_internal/msal_managed_identity_client.py [hunk 3]
```diff
@@ -23,10 +23,13 @@ class MsalManagedIdentityClient(abc.ABC):  # pylint:disable=client-accepts-api-v
     """Base class for managed identity client wrapping MSAL ManagedIdentityClient."""
 
     # pylint:disable=missing-client-constructor-parameter-credential
-    def __init__(self, **kwargs: Any) -> None:
-        self._settings = kwargs
+    def __init__(
+        self, *, client_id: Optional[str] = None, identity_config: Optional[Mapping[str, str]] = None, **kwargs: Any
+    ) -> None:
+        self._client_id = client_id
+        self._identity_config = identity_config
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_internal/msal_managed_identity_client.py:30]: _settings is the same concept of option_bag in get_token_info().

We may pass some arguments into msal.ManagedIdentity in the future.

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_internal/msal_managed_identity_client.py]: Updated to keep the _settings usage

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**